### PR TITLE
Feature/scrum 3761

### PR DESCRIPTION
**Pull request by tlm**
Adds proper support for current state RDAP domain searches.

### CODE REVIEW

File: src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        this.rdapControllerUtil = rdapControllerUtil;
+    }
+
+    @RequestMapping(value="/{handle:.+}", method=RequestMethod.GET)
+    public ResponseEntity<TopLevelObject> domainPathGet(
+        HttpServletRequest request,
+        @PathVariable("handle") String handle)
+    {
+        LOGGER.info("domain GET path query for {}", handle);
```
Comment (GKTheOne) [inline src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java]: Info logging is probably too high here. I'd recommend debug level. (copy-paste comment for the other initial LOGGER.info's)

File: src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * Controller is responsible for dealing with current state RDAP path segments.
+ */
+@RestController
+@RequestMapping("/domain")
+public class DomainRouteController
+{
+    private final static Logger LOGGER = LoggerFactory.getLogger(DomainRouteController.class);
+
+    private RDAPControllerUtil rdapControllerUtil = null;
```
Comment (khloke) [inline src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java]: (Suggestion) Make this final and you wouldn't have to initialise it with null.

**Review by khloke — APPROVED** [review]

**Review by JordanTay — APPROVED** [review]

File: src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java (hunk 2 above)
Comment (tlm) [reply src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java]: Done & Done

File: src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java (hunk 1 above)
Comment (tlm) [reply src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java]: Done & Done

**Review by khloke — APPROVED** [review]

**Review by tomhrr — APPROVED** [review]

**Pull request merged**